go/tasks/plugins/k8s/sidecar/sidecar.go

```diff
@@ -71,11 +70,15 @@ func validateAndFinalizePod(
 type sidecarJob struct {
 	PodSpec              *k8sv1.PodSpec
 	PrimaryContainerName string
+	Annotations          map[string]string
```

Comment:
why do we need this, why can we not use annotations in the PodSpec directly?

Also adding this to struct will not work, it has to be marshalled from flytekit - how does that happen?

Reply:
Annotations/Labels is not part of the `PodSpec`, it is in the `Pod` .

Reply:
So I am planning to open a Pr in flytekit as well to allow annotations/labels for sidecar task.

Reply:
https://github.com/flyteorg/flyteidl/pull/171 is for the IDL change